### PR TITLE
Support diffusion gemma

### PR DESCRIPTION
**Pull request by sudnya**
Support DiffusionGemma inference; fix Tokenformer serving

Adds DiffusionGemma support, which required pulling upstream vLLM changes through our fork. Along the way it fixes Tokenformer adapter serving, which was broken for every model.

- The build never used the fork the Dockerfile named. docker-compose.yaml passed VLLM_REPO/VLLM_BRANCH as explicit build args, and those override Dockerfile ARG defaults. Every build cloned the old fork.
- Serving loaded the wrong checkpoint. Both call sites used sorted(glob("*.pt"))[0] — lexical, so a 2000-step job served checkpoint_1799.pt.

The rest: --enable-tokenformer was never passed after the fork split it from --enable-lora; three v0.26.0 API drifts blocked startup; the trainer was training vision towers; hybrid mode relocated base weights under .base_layer.; and llama/gemma4 both lacked the state_dict override that exposes q/k/v to adapter matching.

DiffusionGemma needed three more: a dtype fix in its sampler, the adapter subsystem disabled (it doesn't declare SupportsLoRA), and sampling-parameter handling — diffusion models require temperature to be exactly 1.0, while ScalarLM and the chat UI both sent 0.

Verified

```
┌───────────────────────────┬────────────────┬───────────────┐
│           Model           │   Inference    │ Train → serve │
├───────────────────────────┼────────────────┼───────────────┤
│ tiny-random/gemma-4-dense │ pass           │ pass          │
├───────────────────────────┼────────────────┼───────────────┤
│ masint/tiny-random-llama  │ pass           │ pass          │
├───────────────────────────┼────────────────┼───────────────┤
│ tiny-random/qwen3         │ pass           │ pass          │
├───────────────────────────┼────────────────┼───────────────┤
│ diffusiongemma-26B-A4B-it │ pass (cluster) │ out of scope  │
└───────────────────────────┴────────────────┴───────────────┘
```

qwen3 is the control  natively supported, no ScalarLM patches  so its passing means the shared pipeline is sound rather than the patched models passing by luck.

enable_lora now defaults to false (matching adapter_type). train_generate.py defaults to 2000 steps; the previous 150 could not converge, so the test could not pass on a healthy system. Base image moved to NGC 26.05. 17 new unit tests.

Known limitations: 
DiffusionGemma is inference-only. Training is out of scope for this PR.

One adapter per server — and it fails silently. Tokenformer adapters are applied by mutating the shared model weights: activate_adapter merges the adapter's tensors into the model's full state dict and calls model.load_weights(...). There are no per-adapter slots, so only one adapter can be resident at a time. Meanwhile the server registers every directory under /app/cray/jobs as an adapter, and set_active_adapters activates each adapter in a batch in turn — last one wins.

With more than one trained model registered they compete over one set of weights, and a request for model A can be answered with model B's. No error, no warning: the log still reports every tensor loaded. The symptom is a model that trains correctly (loss converges) and loads correctly (remapped N, zero dropped) yet returns garbage.

Workaround: one model per server; clear jobs/ between test runs — now more visible since jobs/ is bind-mounted and persists across restarts. Real fix: per-request adapter application in the fork, the way LoRA does it with a slot table and batched kernels — a project, not a patch.

One adapter type per server. LoRA serving is off by default; adapter_type: "lora" jobs silently no-op unless SCALARLM_ENABLE_LORA=true, which in turn breaks Tokenformer checkpoints.

gemma-1, gemma-2 and qwen2-vl can't do train→serve on this fork — they declare neither SupportsTokenformer nor the state_dict override.

Three fixes live as ScalarLM build patches — the state_dict overrides and the DiffusionGemma dtype fix. They belong upstream in vllm_fork.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -40,7 +41,7 @@ services:
     cray-nvidia:
         <<: *cray
         runtime: nvidia
-        restart: unless-stopped
+        #restart: unless-stopped
```
Comment (sudnya) [inline docker-compose.yaml:63]: intentional - for debugging
